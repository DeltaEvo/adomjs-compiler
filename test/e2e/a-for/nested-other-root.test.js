const compileAndCreate = require('../compileAndCreate');

const element = compileAndCreate(`
<template a-for="elem1 of list1">
	<span>{{ elem1 }}</span>
	<span>{{ elem1 }}</span>
	<div>
		<template a-for="elem2 of list2">
			<span>{{ elem2 }}</span>
			<span>{{ elem2 }}</span>
		</template>
	</div>
</template>
`);

it('render', () => {
	element.setState({
		list1: ['Hello', 'World'],
		list2: ['Bonjour', 'Monde']
	});


	expect(element.innerHTML).toMatchSnapshot();
});

it('remove elements', () => {
	element.setState({
		list1: ['Hello'],
		list2: ['Bonjour']
	});

	expect(element.innerHTML).toMatchSnapshot();
});
