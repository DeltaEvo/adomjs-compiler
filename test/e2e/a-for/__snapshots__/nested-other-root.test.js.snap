// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remove elements 1`] = `

<span>
  Hello
</span>
<span>
  Hello
</span>
<div>
  <span>
    Bonjour
  </span>
  <span>
    Bonjour
  </span>
</div>

`;

exports[`render 1`] = `

<span>
  Hello
</span>
<span>
  Hello
</span>
<div>
  <span>
    Bonjour
  </span>
  <span>
    Bonjour
  </span>
  <span>
    Monde
  </span>
  <span>
    Monde
  </span>
</div>
<span>
  World
</span>
<span>
  World
</span>
<div>
  <span>
    Bonjour
  </span>
  <span>
    Bonjour
  </span>
  <span>
    Monde
  </span>
  <span>
    Monde
  </span>
</div>

`;
